Test Failure: io.openliberty.opentracing.internal.test.FATOpentracing.testNested1Sync

RTC 300880

Java 2 security problem

```
testNested1Sync:io.openliberty.opentracing.internal.test.TestAppException
    at io.openliberty.opentracing.internal.test.FATUtilsServer.gatherHttpRequest(FATUtilsServer.java:165)
    at io.openliberty.opentracing.internal.test.FATOpentracing.verifyNestedService1(FATOpentracing.java:1108)
    at io.openliberty.opentracing.internal.test.FATOpentracing.testNested1Sync(FATOpentracing.java:911)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```

```
[8/7/24, 0:18:03:084 UTC] 00000032 nliberty.opentracing.internal.OpentracingJaxRsEMCallbackImpl W CWMOT0009W: The OpenTracing exception mapper detected and is handling an unhandled exception from the JAX-RS application. 
                                                                                                               java.lang.ExceptionInInitializerError
	at java.base/java.lang.J9VMInternals.ensureError(J9VMInternals.java:210)
	at java.base/java.lang.J9VMInternals.recordInitializationFailure(J9VMInternals.java:199)
	at org.apache.cxf.transport.http.asyncclient.AsyncHTTPConduitFactory.createConduit(AsyncHTTPConduitFactory.java:270)
	at org.apache.cxf.transport.http.asyncclient.AsyncHTTPConduitFactory.createConduit(AsyncHTTPConduitFactory.java:261)
	at org.apache.cxf.transport.http.HTTPTransportFactory.getConduit(HTTPTransportFactory.java:233)
	at org.apache.cxf.transport.http.HTTPTransportFactory.getConduit(HTTPTransportFactory.java:215)
	at org.apache.cxf.endpoint.AbstractConduitSelector.createConduit(AbstractConduitSelector.java:144)
	at org.apache.cxf.endpoint.AbstractConduitSelector.getSelectedConduit(AbstractConduitSelector.java:108)
	at org.apache.cxf.endpoint.UpfrontConduitSelector.prepare(UpfrontConduitSelector.java:63)
	at org.apache.cxf.jaxrs.client.ClientConfiguration.prepareConduitSelector(ClientConfiguration.java:109)
	at org.apache.cxf.jaxrs.client.AbstractClient.prepareConduitSelector(AbstractClient.java:914)
	at org.apache.cxf.jaxrs.client.AbstractClient.createMessage(AbstractClient.java:1028)
	at org.apache.cxf.jaxrs.client.WebClient.finalizeMessage(WebClient.java:1111)
	at org.apache.cxf.jaxrs.client.WebClient.doChainedInvocation(WebClient.java:1084)
	at org.apache.cxf.jaxrs.client.WebClient.doInvoke(WebClient.java:932)
	at org.apache.cxf.jaxrs.client.WebClient.doInvoke(WebClient.java:901)
	at org.apache.cxf.jaxrs.client.WebClient.invoke(WebClient.java:461)
	at org.apache.cxf.jaxrs.client.SyncInvokerImpl.method(SyncInvokerImpl.java:135)
	at org.apache.cxf.jaxrs.client.SyncInvokerImpl.method(SyncInvokerImpl.java:130)
	at org.apache.cxf.jaxrs.client.SyncInvokerImpl.get(SyncInvokerImpl.java:50)
	at org.apache.cxf.jaxrs.client.spec.InvocationBuilderImpl.get(InvocationBuilderImpl.java:96)
	at io.openliberty.opentracing.internal.test.service.FATUtilsService.invoke(FATUtilsService.java:40)
	at io.openliberty.opentracing.internal.test.service.FATOpentracingService.getNested(FATOpentracingService.java:449)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:574)
	at com.ibm.ws.jaxrs20.cdi.component.JaxRsFactoryImplicitBeanCDICustomizer.serviceInvoke(JaxRsFactoryImplicitBeanCDICustomizer.java:352)
	at com.ibm.ws.jaxrs20.server.LibertyJaxRsServerFactoryBean.performInvocation(LibertyJaxRsServerFactoryBean.java:642)
	at com.ibm.ws.jaxrs20.server.LibertyJaxRsInvoker.performInvocation(LibertyJaxRsInvoker.java:161)
	at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:101)
	at com.ibm.ws.jaxrs20.server.LibertyJaxRsInvoker.invoke(LibertyJaxRsInvoker.java:276)
	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:213)
	at com.ibm.ws.jaxrs20.server.LibertyJaxRsInvoker.invoke(LibertyJaxRsInvoker.java:460)
	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:112)
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor$1.run(ServiceInvokerInterceptor.java:69)
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor.handleMessage(ServiceInvokerInterceptor.java:120)
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:325)
	at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:129)
	at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:278)
	at com.ibm.ws.jaxrs20.endpoint.AbstractJaxRsWebEndpoint.invoke(AbstractJaxRsWebEndpoint.java:139)
	at com.ibm.websphere.jaxrs.server.IBMRestServlet.handleRequest(IBMRestServlet.java:148)
	at com.ibm.websphere.jaxrs.server.IBMRestServlet.doGet(IBMRestServlet.java:114)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:686)
	at com.ibm.websphere.jaxrs.server.IBMRestServlet.service(IBMRestServlet.java:98)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1266)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:754)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:451)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1362)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1078)
	at com.ibm.ws.webcontainer.servlet.CacheServletWrapper.handleRequest(CacheServletWrapper.java:77)
	at com.ibm.ws.webcontainer40.servlet.CacheServletWrapper40.handleRequest(CacheServletWrapper40.java:87)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:978)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:293)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1260)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:476)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:435)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:569)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:503)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:363)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:330)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:169)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:77)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:516)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:586)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:970)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1059)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:298)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:857)
Caused by: java.security.AccessControlException: Access denied ("java.util.PropertyPermission" "jaxws.http.autoredirect" "read")
	at java.base/java.security.AccessController.throwACE(AccessController.java:177)
	at java.base/java.security.AccessController.checkPermissionHelper(AccessController.java:239)
	at java.base/java.security.AccessController.checkPermission(AccessController.java:386)
	at java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:416)
	at java.base/java.lang.SecurityManager.checkPropertyAccess(SecurityManager.java:1160)
	at java.base/java.lang.System.getProperty(System.java:625)
	at java.base/java.lang.System.getProperty(System.java:608)
	at org.apache.cxf.transport.http.HTTPConduit.<clinit>(HTTPConduit.java:311)
	... 69 more
```

--- a/dev/io.openliberty.opentracing.2.x_fat/publish/servers/opentracingFATServer1/server.xml
+++ b/dev/io.openliberty.opentracing.2.x_fat/publish/servers/opentracingFATServer1/server.xml
@@ -18,4 +18,5 @@
         traceSpecification="*=info:io.openliberty.opentracing*=all:io.openliberty.microprofile.opentracing*=all"
         maxFileSize="100" maxFiles="1" />
 
+    <javaPermission className="java.util.PropertyPermission" name="*" actions="read"/>
 </server>

--- a/dev/io.openliberty.opentracing.2.x_fat/publish/servers/opentracingFATServer2/server.xml
+++ b/dev/io.openliberty.opentracing.2.x_fat/publish/servers/opentracingFATServer2/server.xml
@@ -13,4 +13,5 @@
         traceSpecification="*=info:io.openliberty.opentracing*=all:io.openliberty.microprofile.opentracing*=all"
         maxFileSize="100" maxFiles="1" />
 
+    <javaPermission className="java.util.PropertyPermission" name="*" actions="read"/>
  </server>

--- a/dev/io.openliberty.opentracing.2.x_fat/publish/servers/opentracingFATServer3/server.xml
+++ b/dev/io.openliberty.opentracing.2.x_fat/publish/servers/opentracingFATServer3/server.xml
@@ -16,4 +16,5 @@
         traceSpecification="*=info:io.openliberty.opentracing*=all:io.openliberty.microprofile.opentracing*=all"
         maxFileSize="100" maxFiles="1" />
 
+    <javaPermission className="java.util.PropertyPermission" name="*" actions="read"/>
 </server>

--- a/dev/io.openliberty.opentracing.2.x_fat/publish/servers/opentracingFATServer4/server.xml
+++ b/dev/io.openliberty.opentracing.2.x_fat/publish/servers/opentracingFATServer4/server.xml
@@ -12,4 +12,5 @@
         traceSpecification="*=info:io.openliberty.opentracing*=all:io.openliberty.microprofile.opentracing*=all"
         maxFileSize="100" maxFiles="1" />
 
+    <javaPermission className="java.util.PropertyPermission" name="*" actions="read"/>  
 </server>

